src/vm/appdomain.cpp

```diff
@@ -12956,6 +12956,11 @@ void AppDomain::CreateADUnloadWorker()
         DWORD stackSize = 512 * 1024; // leave X64 unchanged since we have plenty of VM
 #endif
         Thread *pThread = SetupUnstartedThread();
+        if (pThread == NULL)
+        {
+            ThrowOutOfMemory();            
```

Comment:
IMHO it's not just about the # of places that call it. It's about when you are calling new without the nothrow you want to conform to whatever new decides to do instead of throwing an individual exception at each place that calls it (imagine if we changed the exception new throws, yes very unlikely but not unthinkable). The code is cleaner that way. But I don't have a very strong opinion on this.